test/test_plugin_helper.rb

```diff
@@ -6,3 +6,21 @@
 # Add plugin to FactoryGirl's paths
 FactoryGirl.definition_file_paths << File.join(File.dirname(__FILE__), 'factories')
 FactoryGirl.reload
+
+# Foreman's setup doesn't handle cleaning up for Minitest::Spec
+DatabaseCleaner.strategy = :transaction
+
+class Minitest::Spec
+  class << self
+    alias_method :context, :describe
+  end
+
+  before :each do
+    DatabaseCleaner.start
+    Setting::RemoteExecution.load_defaults
+  end
+
+  after :each do
+    DatabaseCleaner.clean
```

Comment:
Usually, unit tests do not touch database, but it's tricky with ActiveRecord so we need things like db cleaner. At least from my experience with mysql, this made a big difference. Anyway I haven't got to testing yet so I have no idea why it is necessary.